File: app/src/main/java/im/fdx/v2ex/network/cookie/CookiePersistor.java
package im.fdx.v2ex.network.cookie;

import java.util.Collection;
import java.util.List;

import okhttp3.Cookie;

/**
 * Created by fdx on 2017/3/16.
 */

public interface CookiePersistor {

    void removeAll(List<Cookie> cookies);

    void clear();

    void persist(Cookie cookie);

    void persistAll(Collection<Cookie> cookies);

    List<Cookie> loadAll();


}

File: app/src/main/java/im/fdx/v2ex/data/network/cookie/CookiePersistor.kt
package im.fdx.v2ex.data.network.cookie

import okhttp3.Cookie

/**
 * Created by fdx on 2017/3/16.
 */

interface CookiePersistor {

    fun removeAll(cookies: List<Cookie>)

    fun clear()

    fun persist(cookie: Cookie)

    fun persistAll(cookies: Collection<Cookie>)

    fun loadAll(): List<Cookie>


}

